base/src/OrderedCacheOfFiles.cpp

```diff
@@ -126,48 +126,79 @@ bool OrderedCacheOfFiles::probe(boost::filesystem::path rootDir, std::string& vi
 	{
 		return false;
 	}
+	std::vector<boost::filesystem::path> dateDir;
+	std::vector<boost::filesystem::path> hourDir;
+	std::vector<boost::filesystem::path> mp4Files;
+	// Note- direction: synced with playback direction
+	fs::directory_iterator dateDirIter(rootDir), dateDirEndIter;
+	LOG_ERROR << "parsing files from dir <" << *dateDirIter << ">";
 
-	boost::filesystem::recursive_directory_iterator dir(rootDir), end;
-
-	LOG_INFO << "parsing files from dir <" << *dir << ">";
-
-	for (dir; dir != end; ++dir)
+	for (dateDirIter; dateDirIter != dateDirEndIter; ++dateDirIter)
 	{
-		LOG_ERROR << "Checking dir <" << dir->path() << ">";
-		auto path = dir->path();
-		if (boost::filesystem::is_directory(dir->path()))
+		if (fs::is_directory(dateDirIter->path()))
 		{
-			auto parentPath = dir->path().parent_path();
+			auto parentPath = dateDirIter->path().parent_path();
 
 			// potential date folder
-			if (boost::filesystem::equivalent(parentPath, rootDir))
+			if (fs::equivalent(parentPath, rootDir))
 			{
-				if (!datePatternCheck(dir->path()))
+				if (datePatternCheck(dateDirIter->path()))
 				{
-					// skip going inside
-					dir.no_push();
+					dateDir.push_back(dateDirIter->path());
 				}
 			}
-			// potential hour folder
-			else
+		}
+
+	}
+
+	if (dateDir.size())
+	{
+		std::sort(dateDir.begin(), dateDir.end());
+	}
+
+	for (auto& dateDirPath : dateDir)
+	{
+		fs::directory_iterator hourDirIter(dateDirPath), hourDirEndIter;
+
+		for (hourDirIter; hourDirIter != hourDirEndIter; ++hourDirIter)
+		{
+			if (fs::is_directory(hourDirIter->path()))
 			{
-				if (!hourPatternCheck(dir->path()))
+				// potential hour folder
+				if (hourPatternCheck(hourDirIter->path()))
 				{
-					// skip going inside
-					dir.no_push();
+					hourDir.push_back(hourDirIter->path());
 				}
 			}
-			// dont add folder paths to relevant files
-			continue;
 		}
+	}
+
+	if (hourDir.size())
+	{
+		std::sort(hourDir.begin(), hourDir.end());
+	}
 
+	for (auto& hourDirPath : hourDir)
+	{
+		fs::directory_iterator mp4FileIter(hourDirPath), mp4FileEndIter;
 		// potential video file
-		if (!filePatternCheck(dir->path()))
+		for (mp4FileIter; mp4FileIter != mp4FileEndIter; ++mp4FileIter)
 		{
-			continue;
+			if (filePatternCheck(mp4FileIter->path()))
+			{
+				mp4Files.push_back(mp4FileIter->path());
+			}
 		}
+	}
 
-		videoName = dir->path().string();
+	if (mp4Files.size())
```

Comment:
instead:
check size, if empty, return false
else, sort and assign first video file